### DISCUSSION TITLE
Improve CI for more different platforms and add more sanitizers

### DISCUSSION THREAD

**Discussion by tigercosmos** [solvcon/modmesh#346] [Ideas]
We can improve the CI by testing with:
- platforms
  - CentOS
  - FreeBDS
  - Android
  - mingw-w64
- Santitizers
  - Address Sanitizer
  - Undefined Behavior Sanitizer
  - Memory Leak Sanitizer

**Comment by yungyuc** [comment]
It may not be hard to add more sanitizers in the CI test runs.  In addition, additional hardware platforms may be nice to have.

**Comment by yungyuc** [comment]
@tigercosmos Do we want to spend time on this in foreseeable future?  If not, I'd like to close this issue.

**Comment by tigercosmos** [comment]
@yungyuc what is your target audience? If not assume users only use Windows, macOS, and Ubuntu, it's better to also test other OSs. In addition, I don't see the drawbacks of having sanitizer in CI?

**Comment by yungyuc** [comment]
> @yungyuc what is your target audience? If not assume users only use Windows, macOS, and Ubuntu, it's better to also test other OSs. In addition, I don't see the drawbacks of having sanitizer in CI?

Adding CI runs costs computing and development resources.  If we do not see concrete value in foreseeable future, it is unlikely for us to spend time on it.

Let's have more discussions about it by turning this into a conversation.

**Comment by tigercosmos** [comment]
Professor Ying-Jer Kao's cluster is using CentOS, which is not supported by modmesh yet. We will need to figure out how to set up modmesh on CentOS/RHEL. For the CI, we may need to use RHEL, since CentOS is already deprecated.
